packages/ubuntu_desktop_installer/lib/pages/active_directory/active_directory_widgets.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:ubuntu_desktop_installer/pages/keyboard_layout/keyboard_layout_detector.dart';
+import 'package:ubuntu_wizard/widgets.dart';
+
+import '../../l10n.dart';
+import 'active_directory_l10n.dart';
+import 'active_directory_model.dart';
+
+class DomainNameFormField extends StatelessWidget {
+  const DomainNameFormField({super.key, this.fieldWidth});
+
+  final double? fieldWidth;
+
+  @override
+  Widget build(BuildContext context) {
+    final lang = AppLocalizations.of(context);
+    final domainName = context.select((ActiveDirectoryModel m) => m.domainName);
+    final validation =
+        context.select((ActiveDirectoryModel m) => m.domainNameValidation);
+
+    return ValidatedFormField(
+      autofocus: true,
+      fieldWidth: fieldWidth,
+      labelText: lang.activeDirectoryDomainLabel,
+      initialValue: domainName,
+      successWidget: const SuccessIcon(),
+      validator: CallbackValidator(
+        (_) =>
+            validation == null ||
+            validation.singleOrNull == AdDomainNameValidation.OK,
+        errorText: validation?.firstOrNull?.localize(lang) ?? '',
+      ),
+      onChanged: context.read<ActiveDirectoryModel>().setDomainName,
```

Comment:
This worries me a little bit because when we set the domain name in the mode, we call for validation, and if the string is valid, then a network call is made to confirm the domain is reachable. It's usually fast in the positive scenario, but can easily span for more than 5 seconds if the domain name is not found. Chances are we won't see that often, because subiquity provides you with domain auto discovery, so a user requesting to join AD might not even need to type the domain name, in such case we should be fine.

Reply:
Hmm, I'd also feel more confident about this if we could split the basic regexp-based input validation and connection testing to separate endpoints. Or is it already possible with the latest API?

Reply:
I'll take care of that split in subiquity and ping you back once done.

Reply:
I've updated it to ping the domain controller on editing complete or when pressing _Test connection_ 

![image](https://user-images.githubusercontent.com/140617/223418776-8fdef996-f1bc-4a0a-906d-67ec04b5b1a9.png)